internal/app/usercache/usercacheimp.go

```diff
@@ -0,0 +1,50 @@
+package usercache
+
+import (
+	"github.com/TheFlies/ofriends/internal/app/types"
+	"github.com/globalsign/mgo"
+	"github.com/globalsign/mgo/bson"
+	"github.com/pkg/errors"
+)
+
+type (
+	UsercacheMongoReponsitory struct {
+		session *mgo.Session
+	}
+)
+
+func NewUsercachMongoReopnsitoty(s *mgo.Session) *UsercacheMongoReponsitory {
+	return &UsercacheMongoReponsitory{
+		session: s,
+	}
+}
+func (r *UsercacheMongoReponsitory) FindUserByUserName(username string) (*types.Usercache, error) {
+	s := r.session.Clone()
+	defer s.Close()
+	var user *types.Usercache
+	if err := r.collection(s).Find(bson.M{"username": username}).One(&user); err != nil {
+		return nil, errors.Wrap(err, "Fail to find username in database")
```

Comment:
error can be wrapped in multiple level hence should not be started with CAP character..